synapse/cores/lmdb.py

```diff
@@ -77,28 +77,9 @@
 def _round_up(val, modulus):
     return val - val % -modulus
 
-class DatabaseInconsistent(Exception):
-    ''' If you get this Exception, that means the database is corrupt '''
-    pass
-
-class DatabaseLimitReached(Exception):
-    ''' You've reached some limit of the database '''
-    pass
-
-# Python 2.7 version of lmdb buffers=True functions return buffer objects.  Python 3 version returns
-# memoryview objects
-if sys.version_info > (3, 0):
-    def _memToBytes(x):
-        return x.tobytes()
-else:
-    def _memToBytes(x):
-        return str(x)
-
-    def lru_cache(maxsize):
-        ''' A dumb passthrough.  Python 2 impl is just going to be a tad slower '''
-        def actual_decorator(wrappee):
-            return wrappee
-        return actual_decorator
+# Python 2.7 version of lmdb buffers=True functions return buffer objects.
+# Python 3 version returns memoryview objects
+memToBytes = s_compat.memToBytes
```

Comment:
this feels a hair inconsistent ( since we're also accessing members from s_compat below.  I think i'd rather we either go with 
```
s_compat.memTyBytes()
```
calls below, or change our compat import to
```
from synapse.compat import lru_cache, memToBytes, etcetcetc
```